firebase-appdistribution/test-app/src/main/java/com/googletest/firebase/appdistribution/testapp/CustomNotificationFeedbackTrigger.kt

```diff
@@ -6,204 +6,113 @@ import android.app.*
-    }
-    isEnabled = false
-    activityToScreenshot = null
-  }
-
-  private fun showNotification(context: Context) {
-    if (ContextCompat.checkSelfPermission(context, POST_NOTIFICATIONS) == PERMISSION_DENIED) {
-      Log.w(TAG, "Not showing notification because permission has not been granted.")
-      return
-    }
-
-    val intent = Intent(context, TakeScreenshotAndTriggerFeedbackActivity::class.java)
-    intent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY)
-    val pendingIntent =
-      PendingIntent.getActivity(
-        context,
-        /* requestCode = */ 0,
-        intent,
-        PendingIntent.FLAG_IMMUTABLE
-      )
-    val builder =
-      NotificationCompat.Builder(context, FEEDBACK_NOTIFICATION_CHANNEL_ID)
-        .setSmallIcon(R.mipmap.ic_launcher)
-        .setContentTitle(context.getText(R.string.feedbackTriggerNotificationTitle))
-        .setContentText(context.getText(R.string.feedbackTriggerNotificationText))
-        .setPriority(NotificationCompat.PRIORITY_HIGH)
-        .setContentIntent(pendingIntent)
-    val notificationManager = NotificationManagerCompat.from(context)
-    Log.i(TAG, "Showing notification")
-    notificationManager.notify(FEEDBACK_NOTIFICATION_ID, builder.build())
-  }
-
-  private fun cancelNotification(context: Context) {
-    val notificationManager = NotificationManagerCompat.from(context)
-    Log.i(TAG, "Cancelling notification")
-    notificationManager.cancel(FEEDBACK_NOTIFICATION_ID)
-  }
-
-  override fun onActivityResumed(activity: Activity) {
-    if (isEnabled) {
-      if (activity !is TakeScreenshotAndTriggerFeedbackActivity) {
-        Log.d(TAG, "setting current activity")
-        activityToScreenshot = activity
+  fun cancelNotification() {
+    synchronized(this) {
+      activityToScreenshot?.let {
+        Log.i(TAG, "Cancelling notification")
+        NotificationManagerCompat.from(it).cancel(FEEDBACK_NOTIFICATION_ID)
       }
     }
   }
-
-  override fun onActivityDestroyed(activity: Activity) {
-    if (activity == activityToScreenshot) {
-      Log.d(TAG, "clearing current activity")
-      activityToScreenshot = null
-    }
-  }
-
-  // Other lifecycle methods
-  override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) {}
-  override fun onActivityStarted(activity: Activity) {}
-  override fun onActivityPaused(activity: Activity) {}
-  override fun onActivityStopped(activity: Activity) {}
-  override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) {}
 }
 
-class TakeScreenshotAndTriggerFeedbackActivity : Activity() {
+class CustomNotificationTakeScreenshotActivity : Activity() {
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
-    val activity = CustomNotificationFeedbackTrigger.activityToScreenshot
-    if (activity == null) {
-      Log.e(TAG, "Can't take screenshot because activity is unknown")
-      return
+    synchronized(CustomNotificationFeedbackTrigger) {
```

Comment:
Why? Also, I think this synchronizes on the class - not the instance like the other two `synchronize` blocks.

Maybe this is an attempt to somehow protect modifications to `activityToScreenshot`, but I think read/writes to reference variables are [atomic](https://docs.oracle.com/javase/tutorial/essential/concurrency/atomic.html). 

The assignment to the local variable `activity` on line 109 makes sure that the value doesn't change between the `null` check on line 110 and line 114. Inlining `CustomNotificationFeedbackTrigger.activityToScreenshot` without synchronization would indeed be a problem.

So I think we should add a comment to the assignment on line 109 talking about this issue.